### PR TITLE
Redesign testing structure of behat to make features more ubiquitous

### LINKED ISSUE #340: make behat scenarios more granular
### Description

In order to do "real" BDD, we should try to step away from a technical language in behat to a specification language which describes workflows of how to receive data (the ReactJS frontend is basically just a UI for those workflows defined as API).

To achieve that, some last steps are necessary to achieve this.
### Tasks
- [x] create a project root dir for the features (`features/`) as it's rather a description and not something technical
- [x] rename subdirs:

```
- api
  - {resource}
    - {feature}
- data
  - {resource}
    - {feature}
```
- [x] split files to segregate every feature (also on the data layer)
- [x] checkout whether more granular steps are possible
- [x] test REST APIs directly (this avoids useless feature contexts)

### PR DESCRIPTION
**Pull request by Ma27**
### Overview

- redesigned behat features
- added `ApiContext`
- simplified integration tests

### Related issues

resolves #340

### CODE REVIEW

**Comment by Ma27** [comment]
@benbieler ok, it's ready to review now.
Fixed this odd composer behavior in the VM, but we might look for a better solution somehow in the next time :)

File: src/AppBundle/Tests/Acceptance/ApiContext.php [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+     * Evaluates a property path.
+     *
+     * @param string $propertyPath
+     *
+     * @return mixed
+     */
+    private function evaluatePropertyPath(string $propertyPath)
+    {
+        static $accessor = null;
+        if (!$accessor) {
```
Comment (Ma27) [inline src/AppBundle/Tests/Acceptance/ApiContext.php]: unnecessary actually

File: src/AppBundle/Tests/Acceptance/AbstractIntegrationContext.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        return $tester;
+    }
+
+    /**
+     * Simple shortcut to get the entity manager.
+     *
+     * @return EntityManagerInterface
+     */
+    protected function getEntityManager():EntityManagerInterface
```
Comment (Ma27) [inline src/AppBundle/Tests/Acceptance/AbstractIntegrationContext.php]: space between `:` and type hint

File: src/AppBundle/Tests/Acceptance/ApiContext.php [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        $this->client->request('POST', '/api/api-key.json', ['login' => $username, 'password' => $password]);
+        $response = $this->client->getResponse();
+
+        Assertion::eq(200, $response->getStatusCode(), sprintf(
+            'Authentication with username "%s" and password "%s" failed!',
+            $username,
+            $password
+        ));
+
+        $this->apiKey = json_decode($response->getContent(), true)['apiKey'];
```
Comment (Ma27) [inline src/AppBundle/Tests/Acceptance/ApiContext.php]: use `$this->decode($json)` here (might be more readable than `json_decode(..., true)` and covers edge cases with a broken response

File: src/AppBundle/Tests/Acceptance/ApiContext.php [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    }
+
+    /**
+     * @When /^I submit a request to "(.*)"$/
+     *
+     * @param string $request
+     */
+    public function submitRequest(string $request)
+    {
+        $method = ($matched = (bool) preg_match('/^([\w]+) (\/(.*))$/', $request, $matches)) ? $matches[1] : 'GET';
```
Comment (Ma27) [inline src/AppBundle/Tests/Acceptance/ApiContext.php]: might be a private method (just for readability)

File: src/AppBundle/Tests/Acceptance/Functional/User/OnlineUsersContext.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        /** @var \AppBundle\Service\Doctrine\Repository\UserRepository $repository */
+        $repository = $this->getContainer()->get('app.repository.user');
+
+        Assertion::eq(
+            $this->apiContext->getResponse(),
+            array_combine(
+                array_map(function (array $data) use ($repository):string {
+                    return $repository->findOneBy(['username' => $data['username']])->getId();
+                }, $node->getHash()),
+                array_fill(0, 1, 1)
```
Comment (Ma27) [inline src/AppBundle/Tests/Acceptance/Functional/User/OnlineUsersContext.php]: as it means `the following users`, there should be a more generic way than doing `array_fill()` for a single element.

**Review by Ma27 — COMMENTED** [review]
some minor design-related things

**Review by benbieler — APPROVED** [review]
